lib/client.js

```diff
@@ -740,6 +740,7 @@ class Parser {
     if (!request) {
       return -1
     }
+    request.onMessageBegin()
```

Comment:
If we are following the path of extending the request, I believe we should sync this with the `Dispatcher` API and document it accordingly, as it opens a new API.

Reply:
Hm, good point. According to [spec](https://fetch.spec.whatwg.org/#http-network-fetch) we want to "Set timingInfo’s [final network-response start time](https://fetch.spec.whatwg.org/#fetch-timing-info-final-network-response-start-time) to the [coarsened shared current time](https://w3c.github.io/hr-time/#dfn-coarsened-shared-current-time) given fetchParams’s [cross-origin isolated capability](https://fetch.spec.whatwg.org/#fetch-params-cross-origin-isolated-capability), _immediately after the user agent’s HTTP parser receives the first byte of the response_ (e.g., frame header bytes for HTTP/2 or response status line for HTTP/1.x).". 
So I think this is the closest we get with the current llhttp API. Unless there are some alternative hooks where we can add this timing info, without opening a new API?

Reply:
Isn't onHeaders sufficient in a practical sense?

Reply:
It would be close, I guess it depends on how closely we want to adhere to the spec.

I do think it's unfortunate that responseStart (which is time to first byte) is dependent on header size. 
Having accurate responseStart and responseEnd values would also be good for benchmarking message parsing.

Reply:
onResponseStarted?

Reply:
I was also considering if it would be better to call this from outside of the llhttp hooks, as it doesn't really depend on the payload parsing. But I couldn't find an obvious place to hook in, that would also work with the pause/receive functionality. If I understand correctly llhttp keeps track of all that.

Reply:
I think where you have it is good enough for any practical case.